File: ney/level1/행렬곱셈.java
class Solution {
    public int[][] solution(int[][] arr1, int[][] arr2) {
        
        int x = arr1.length;
		int y = arr2[0].length;
		
		int xx = arr1[0].length;
		
		int[][] answer = new int[x][y];
		
		for(int i = 0 ; i < x; i++) {
			for(int j = 0; j < y; j++) {
				for(int n = 0; n <xx ; n++) {
						answer[i][j] 
								+= arr1[i][n] 
										* arr2[n][j];
				}
			}
		}
        return answer;
    }
}

File: kimjeonggi/행렬곱셈.cpp
#include <string>
#include <vector>
#include<iostream>
using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr1, vector<vector<int>> arr2) {
    vector<vector<int>> answer;
    int x = arr1.size();
    int y = arr2[0].size();
    int z = arr1[0].size();
    for(int i=0; i<x; i++){
        vector<int> v;
        for(int j=0; j<y; j++){
            int sum = 0;
            for(int k=0; k<z; k++){
                sum  += arr1[i][k] * arr2[k][j];
            }
            v.push_back(sum);
        }
        answer.push_back(v);
    }
    return answer;
}
